dropshot_endpoint/src/lib.rs

```diff
@@ -127,32 +127,102 @@ pub fn channel(
 ///
 /// The `#[dropshot::api_description]` macro accepts these arguments:
 ///
-/// * `context`: The type of the context on the trait. Optional, defaults to
-///   `Self::Context`.
-/// * `module`: The name of the support module. Optional, defaults to the
-///   `snake_case` version of the trait name.
+/// * `context`: The type of the context on the trait. _Optional, defaults to
+///   `Self::Context`._
+/// * `module`: The name of the support module. _Optional, defaults to the
+///   `snake_case` version of the trait name._
+/// * `tag_config`: Trait-wide tag configuration. _Optional._ For more
+///   information, see [_Tag configuration_](#tag-configuration) below.
 ///
-/// ## Limitations
+/// ### Example: specify a custom context type
 ///
-/// Currently, the `#[dropshot::api_description]` macro is only supported in
-/// module contexts, not function definitions. This is a Rust limitation -- see
-/// [Rust issue #79260](https://github.com/rust-lang/rust/issues/79260) for more
-/// details.
+/// ```ignore
+/// use dropshot::{RequestContext, HttpResponseUpdatedNoContent, HttpError};
+///
+/// #[dropshot::api_description { context = MyContext }]
+/// trait MyTrait {
+///     type MyContext;
+///
+///     #[endpoint {
+///         method = PUT,
+///         path = "/test",
+///     }]
+///     async fn put_test(
+///         rqctx: RequestContext<Self::MyContext>,
+///     ) -> Result<HttpResponseUpdatedNoContent, HttpError>;
+/// }
+/// # // defining fn main puts the doctest in a module context
+/// # fn main() {}
+/// ```
+///
+/// ### Tag configuration
+///
+/// With Dropshot, individual endpoints can have OpenAPI tags associated with
+/// them. For these tags, trait-wide configuration can be specified using the
+/// `tag_config` argument to `api_description`.
```

Comment:
```suggestion
/// Endpoints can have tags associated with
/// them that appear in the OpenAPI document. These provide a means of organizing an API. 
Use the
/// `tag_config` argument to `api_description` to specify tag information for the trait.
```

I think this might be clearer? Feel free to ignore or adapt.